plugins/modules/cloud/misc/proxmox_instance_actions.py

```diff
@@ -0,0 +1,307 @@
+                if status['exitstatus'] == 'OK':
+                    time.sleep(1)
+                    return True
+                else:
+                    raise Exception('Task exited in error -- %s' % status['exitstatus'])
+            timeout = timeout - 1
+            if timeout == 0:
+                raise Exception('Reached timeout while waiting for task. Last line in the log: %s'
+                                % self.lastlog())
+            time.sleep(1)
+
+
+def proxmox_instance_actions_argument_spec():
+    return dict(
+        action=dict(type='str',
+                    required=True,
+                    choices=['reboot', 'reset', 'resume', 'shutdown', 'start', 'stop', 'suspend']
+                    ),
+        force=dict(type='bool', default=False),
+        name=dict(type='str'),
+        node=dict(type='str', required=True),
+        timeout=dict(type='int', default=30),
+        vmid=dict(type='int'),
+    )
+
+
+def main():
+    module_args = proxmox_auth_argument_spec()
+    proxmox_instance_actions_args = proxmox_instance_actions_argument_spec()
+    module_args.update(proxmox_instance_actions_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_one_of=[('api_password', 'api_token_id'), ('name', 'vmid')],
+        required_together=[('api_token_id', 'api_token_secret')],
+        supports_check_mode=False
+    )
+    result = dict(
+        changed=False
+    )
+
+    if not HAS_PROXMOXER:
+        module.fail_json(msg=missing_required_lib('proxmoxer'), exception=PROXMOXER_IMP_ERR)
+
+    proxmox = ProxmoxInstanceActionsAnsible(module)
+    action = module.params['action']
+    force = module.params['force']
+    timeout = module.params['timeout']
+    if module.params['vmid']:
+        vm = proxmox.get_instance(vmid=module.params['vmid'])
+    else:
+        vm = proxmox.get_instance(name=module.params['name'])
+
+    # State and actions matrice:
+    #            +---------+---------+---------+-----------+
+    #            | running | paused  | stopped | prelaunch |
+    # +----------+---------+---------+---------+-----------+
+    # | reboot   |      OK |   Error |   Error |           |
+    # | reset    |      OK |      OK |   Error |           |
+    # | resume   |    Noop |      OK |   Error |           |
+    # | shutdown |      OK |   Error |    Noop |           |
+    # | start    |    Noop |   Error |      OK |           |
+    # | stop     |      OK |      OK |    Noop |        OK |
+    # | suspend  |      OK |    Noop |   Error |           |
+    # +----------+---------+---------+---------+-----------+
+
+    # Skip noop operations directly
+    status = vm.status()
+    if ((status['qmpstatus'] == 'paused' and action == 'suspend')
+       or (status['qmpstatus'] == 'running' and (action == 'resume' or action == 'start'))
+       or (status['qmpstatus'] == 'stopped' and (action == 'stop' or action == 'shutdown'))):
+        module.exit_json(**result)
+
+    try:
```

Comment:
For check mode, probably something like:
```suggestion
    if module.check_mode:
        result['changed'] = True
        module.exit_json(**result)

    try:
```

Reply:
Yes, thank you :)